fml/logging.cc

```diff
@@ -80,9 +84,35 @@ LogMessage::~LogMessage() {
       priority = ANDROID_LOG_FATAL;
       break;
   }
-  __android_log_write(priority, "flutter", stream_.str().c_str());
-#elif defined(OS_IOS)
-  syslog(LOG_ALERT, "%s", stream_.str().c_str());
+  __android_log_write(priority, "flutter", chars);
+#elif defined(__APPLE__)
```

Comment:
Please stick to the macros in `build_config.h`. `OS_MACOSX` covers both iOS and MacOS so that may be used here instead of depending on a macro we don't control.